app/src/main/java/net/squanchy/favorites/view/FavoritesAdapter.kt

```diff
@@ -1,136 +1,71 @@
 package net.squanchy.favorites.view
 
 import android.content.Context
-import android.support.v7.widget.RecyclerView
+import android.support.v7.recyclerview.extensions.ListAdapter
+import android.support.v7.util.DiffUtil
 import android.view.LayoutInflater
 import android.view.ViewGroup
 import net.squanchy.R
 import net.squanchy.schedule.domain.view.Event
-import net.squanchy.schedule.domain.view.Schedule
-import net.squanchy.schedule.domain.view.SchedulePage
-import net.squanchy.schedule.view.EventItemView
-import net.squanchy.schedule.view.EventViewHolder
-import net.squanchy.search.view.HeaderViewHolder
-import org.joda.time.DateTimeZone
-import org.joda.time.LocalDate
 
-internal class FavoritesAdapter(context: Context) : RecyclerView.Adapter<RecyclerView.ViewHolder>() {
+internal class FavoritesAdapter(
+    context: Context
+) : ListAdapter<FavoritesItem, FavoritesViewHolder<*>>(DiffCallback) {
 
     companion object {
         private const val VIEW_TYPE_TALK: Int = 1
         private const val VIEW_TYPE_HEADER: Int = 2
     }
 
-    private val layoutInflater = LayoutInflater.from(context)
-
-    private var schedule = Schedule(emptyList(), DateTimeZone.UTC)
+    lateinit var favoriteClickListener: OnFavoriteClickListener
 
-    private var listener: ((Event) -> Unit)? = null
+    private val layoutInflater = LayoutInflater.from(context)
 
     init {
         setHasStableIds(true)
     }
 
-    override fun getItemId(position: Int): Long = produceData(
-        pages = schedule.pages,
-        absolutePosition = position,
-        headerProducer = { schedulePage -> (-schedulePage.dayId.hashCode()).toLong() },
-        rowProducer = { schedulePage, positionInPage -> schedulePage.events[positionInPage].numericId }
-    )
-
-    fun updateWith(schedule: Schedule, listener: (Event) -> Unit) {
-        this.schedule = schedule
-        this.listener = listener
-        notifyDataSetChanged()
+    override fun getItemId(position: Int): Long {
+        return getItem(position).id
```

Comment:
maybe inline this one?

Reply:
1af1597dc66429bb291d6337bc8f818fea746605